pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -213,3 +225,25 @@ func noResponseHints(err error) string {
 
 	return ""
 }
+
+func clientWithOneRedirects() *http.Client {
+	clientCustom := &http.Client{}
+	clientCustom.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+	return clientCustom
+}
+
+// See if the URL is redirected to another URL, and return the status code of the redirection
+func sendHTTPHEADRequestToEndpoint(url string, client *http.Client) (int, error) {
+	res, err := client.Head(url)
+	if err != nil {
+		return -1, err
+	}
+	defer res.Body.Close() // Close the response body
+	// Since the flare only accept post, when we perform a head request, we should expect a redirection before the final response
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏This function is not specific to flare so it shouldn't mention it, you can just explain what you expect, eg:
```suggestion
	// Expected status codes are OK or a redirection
```

Reply:
thx, fixed too